packages/@ember/-internals/metal/lib/tracked/debugging.ts

```diff
@@ -0,0 +1,110 @@
+import { Tag, CONSTANT_TAG, UpdatableTag } from '@glimmer/reference';
+
+import { Tracker } from './tracker';
+
+type Option<T> = T | null;
+
+Ember.EMBER_DEBUG = {};
+
+let TAG_MAP = new Map<Tag | UpdatableTag, any>();
+
+interface TrackerHistory {
+  property: string;
+  value: any;
+  tracker: Tracker;
+}
+
+interface DebugTracking {
+  history: TrackerHistory[];
+  stacks: Set<Tracker>;
+}
+
+interface TrackerSnapshot {
+  tags: Array<Tag | UpdatableTag>;
+}
+
+interface TrackerProperty {
+  propertyName: string;
+  objectName: string;
+  objectRef: object;
+  tag: Tag | UpdatableTag;
+}
+
+Ember.EMBER_DEBUG.TRACKING = {
+  history: [],
+  stacks: new Set<Tracker>(),
+} as DebugTracking;
+
+function normalizeTrackerEntry(tag: any) {
+  // it looks like there are two types of data on a tracker's tag list.
+  // one type has the shape (which maybe isn't a tag?):
+  //   key: string
+  //   ref: object
+  //   tag: object
+  // the other is an actual Tag,
+  //   but with a private secret property _propertyName
+
+  let hostObject = TAG_MAP.get(tag);
+
+  let result: Partial<TrackerProperty> = {
+    objectName: hostObject && hostObject.__proto__.constructor.name,
+    objectRef: hostObject,
+    tag,
+  };
+
+  result.propertyName = tag.key || tag._propertyKey;
+
+  return result;
+}
+
+function prettyPrintTrackingInfo() {
+  let history = Ember.EMBER_DEBUG.TRACKING.history;
+
+  console.log(history, TAG_MAP);
+  history.forEach((tracker: TrackerSnapshot, i: number) => {
+    // tags seems to always be a touple
```

Comment:
I don't know if this is always true.

In the app I'm debugging this with, this is my property dep tree:

```
count
|- doubled
|- <ApplyMath @num={{this.count}} />
   |- dividedByThree
      |- again
```

So, indirectly, the property `again` should have a stack of 3 high (but it doesn't?)?

count -> dividedByThree -> again